notifications/v1/types.go

```diff
@@ -0,0 +1,16 @@
+package v1
+
+// ⚠️ WARNING: These types MUST match the SAMS implementation, at
+// backend/internal/notification/types.go
+
+const (
+	nameUserDeleted = "UserDeleted"
+)
```

Comment:
Should these be exported, and then the SAMS implementation can reference them when publishing events?

Reply:
I actually prefer try not to do that, it sounds very reasonable thing to do but in my experience, it's a nasty dependency to deal with (i.e. have to make a PR on this repo just for a string const, then go mod replace on the SAMS repo with a broken PR. This could totally be Go's toolchain fault, but I don't feel strong that it's worth it doing it now)

Reply:
I mean string const addition is basically an API addition, i.e. "this is coming soon", and is detached from implementation, like any proposed change to the API spec. WDYT?

Reply:
(Apologies in advance if I'm totally missing something here.)

I'd be inclined to avoid the circular dependency between the SAMS SDK and SAMS itself, as it does seem like it would complicate things for little-to-know benefit.

Ultimately consumers of the SAMS SDK _shouldn't_ know anything about the topic names, because it's irrelevant to their needs. As someone hooking into this, I just need to be concerned about using the `type ReceiveHandlers struct`. And registering the appropriate `OnEventX` handlers. The actual names used in the Pub/Sub topics is 100% irrelevant.

> I mean string const addition is basically an API addition, i.e. "this is coming soon", and is detached from implementation, like any proposed change to the API spec. WDYT?

If SAMS starts or stops submitting events with a new name, clients don't really care. It should be a warning-level log, since it indicates that the client is running on an older build of the SAMS SDK, but there being events that are in-the works doesn't really mean much.

i.e. as a consumer of this library, either I provide an `OnEventX` handler to start processing those events, or not. The current state of rollout on the SAMS-end isn't relevant?

Reply:
Fair enough, but `⚠️ WARNING: These types MUST match the SAMS implementation, at X`-type comments feels like an uncomfortable invariant for me, irrespective of "does it matter or not".

Also, for "clients" in terms of "user calling the SDK" indeed they shouldn't care, but for "clients" in terms of "the implementation of the SDK being a client to SAMS", it certainly does care about what gets published, which is why this is here in the first place

Reply:
> Fair enough, but `⚠️ WARNING: These types MUST match the SAMS implementation, at X`-type comments feels like an uncomfortable invariant for me, irrespective of "does it matter to clients or not".

We can reword this and change it to a note instead to something like "see xxxx for the full list of supported notifications". Does that make you more comfortable? 😁

Reply:
I mean the implication is still a "we should match some hand-maintained invariant", but sure, not a huge deal